holmes/plugins/toolsets/coralogix/toolset_coralogix_logs.py

```diff
@@ -73,84 +78,104 @@ def __init__(self, toolset: BaseCoralogixToolset):
         if namespace_name:
-            query_filters.append(
-                f"{self.toolset.config.labels.namespace}:{namespace_name}"
-            )
+            query_filters.append(f"{config.labels.namespace}:{namespace_name}")
         if pod_name:
-            query_filters.append(f"{self.toolset.config.labels.pod}:{pod_name}")
+            query_filters.append(f"{config.labels.pod}:{pod_name}")
         if app_name:
-            query_filters.append(f"{self.toolset.config.labels.app}:{app_name}")
-        if min_log_level:
-            min_severity = LOG_LEVEL_TO_SEVERITY.get(
-                min_log_level.upper(), 2
-            )  # Default to INFO (2) if not found
-            query_filters.append(f"coralogix.metadata.severity:[{min_severity} TO *]")
+            query_filters.append(f"{config.labels.app}:{app_name}")
 
         query_string = " AND ".join(query_filters)
+        query_string = f"source logs | lucene '{query_string}' | limit {log_count}"
+        return query_string
 
+    def _invoke(self, params: Any) -> StructuredToolResult:
+        if not self.toolset.config:
+            return StructuredToolResult(
+                status="success",
+                error="The coralogix/logs toolset is not configured",
+                data=None,
+                params=params,
+            )
+
+        (start, end) = process_timestamps_to_rfc3339(
+            start_timestamp=params.get("start"),
+            end_timestamp=params.get("end"),
+            default_time_span_seconds=DEFAULT_TIME_SPAN_SECONDS,
+        )
+        query_string = self._get_query_string(self.toolset.config, params)
         query = {
-            "query": f"source logs | lucene '{query_string}' | limit {log_count}",
+            "query": query_string,
             "metadata": {
                 "syntax": "QUERY_SYNTAX_DATAPRIME",
                 "startDate": start,
                 "endDate": end,
             },
         }
-
         response = execute_query(
-            base_url=self.toolset.config.base_url,
+            domain=self.toolset.config.domain,
             api_key=self.toolset.config.api_key,
             query=query,
         )
 
         # Do not print tags if they are repeating the query
+        namespace_name = params.get("namespace_name", None)
+        pod_name = params.get("pod_name", None)
         add_namespace_tag = not namespace_name and not pod_name
         add_pod_tag = not pod_name
 
         try:
-            return format_logs(
+            logs = format_logs(
                 raw_logs=response.text.strip(),
                 add_namespace_tag=add_namespace_tag,
                 add_pod_tag=add_pod_tag,
             )
+            return StructuredToolResult(
+                status="success" if logs else "no_data",
+                error=None if logs else "The query returned no logs",
+                data=logs,
+                query=query_string,
+                url=build_coralogix_link_to_logs(
```

Comment:
Just to make sure, it is safe to run build_coralogix_link_to_logs if there is no data?

Reply:
Yes, it's actually one of the main use cases for having a URL